### DISCUSSION TITLE
What is the correct way to use this library with NestJS? I get no types

### DISCUSSION THREAD

**Discussion by ashley-jisc** [typegoose/typegoose#697] [Q&A]
For the purposes of this example, in this screenshot, I have two entities (cat & kitten).
Cat is using Typegoose and Kitten is using Prisma. 

If you see, there is no type error when I attempt to put an integer value for name on the cat with Typegoose, but I do get an error as you would expect with Prisma since the `name` field is set to a type of `string`. 

Is this how Typegoose is supposed to work, or should I be getting a type error as seen with Prisma? 
What am I doing wrong here?
<img width="1325" alt="image" src="https://user-images.githubusercontent.com/85339838/164779408-0d3b99e8-6c10-49a3-bdf6-c9c80e547a89.png">

Code: 
cat.service.ts
```javascript
import { Injectable } from '@nestjs/common';
import { Cat } from './cat.schema';
import { getModelForClass } from '@typegoose/typegoose';

@Injectable()
export class CatService {

    async create(): Promise<Cat> {
        const catModel = getModelForClass(Cat);
        const newCat = await catModel.create({
            name: 123, // <-- no type error?
        });

        return newCat.save();
    }
}
```
cat.schema.ts
```javacsript
import { Prop } from '@typegoose/typegoose';

export class Cat {
    @Prop()
    name: string;

    @Prop()
    age: number;

    @Prop()
    breed: string;
}
```

**Comment by smolinari** [comment]
@ashley-jisc - check out this package. 

https://www.npmjs.com/package/@m8a/nestjs-typegoose

Scott

**Reply by ashley-jisc** [reply]
Hi Scott, thanks for the suggestion. I have just tried to use it, but the same problem still persists.

**Comment by smolinari** [comment]
And did you inject the model? Can you make a minimum reproduction repo?

Scott

**Reply by ashley-jisc** [reply]
Sure - please see the cat service
https://github.com/proevilz/ts-demo

**Reply by smolinari** [reply]
I found this in the Typegoose docs. 

https://typegoose.github.io/typegoose/docs/guides/faq/#why-does-new-model-not-have-types

Scott

**Reply by ashley-jisc** [reply]
Ah, that explains it. Thanks.

**Comment by hasezoey** [comment]
it may be that mongoose's types dont enforce these types strictly in `new Model(here)` or in `Model.create(here)`, but i currently cannot test your repro-repo (not on the right maschine, will do later)

also here are some things to note:
- i would generally not recommend to use what `.save` returns as a return value, just do `await doc.save(); return doc;`
- your return type does not quite match up with what is actually returned, currently it is set to `Promise<Class>`, but it should be `Promise<DocumentType<Class>>` (not having this set could end up in weird situations)
- i also would not recommend to use `getModelForClass` each time you want the model, either use NestJS's injection system (via `nestjs-typegoose`), or export the model as a value (like directly from where the class is or wherever you want)*1

*1 using `getModelForClass` firstly is less performance than doing it once and using exports (has to process more stuff) and some other things that may or may not cause problems when internal things get changed

**Comment by findingtheconcept** [comment]
Hi! Please do not take this as a spam, but now I'm looking for a people who will try my own `NestJS` wrapper for `Typegoose` that I recently created. So I found this topic and decided to left here this message.

Here is the repository of my library: https://github.com/GrapeoffJS/kindagoose

I described my motives for creating a new wrapper in documentation. Hope you'll like it!
